File: frontend/angular-book-ecommerce/src/app/common/update-user-dto.ts
export class UpdateUserDTO {
    username: string;    
    password: string;    
    retype_password: string;    
    phone_number: string;
    facebook_account_id: number;
    google_account_id: number;
    email: string;
    date_of_birth: string;
   
    
    constructor(data: any) {
        this.username = data.username;
        this.password = data.password;
        this.retype_password = data.retype_password;
        this.phone_number = data.phone_number;
        this.facebook_account_id = data.facebook_account_id;
        this.google_account_id = data.google_account_id;
        this.email = data.email;
        this.date_of_birth = data.date_of_birth;
    }
}

File: backend/spring-boot-book-ecommerce/src/main/java/com/luvannie/springbootbookecommerce/dto/UpdateUserDTO.java
package com.luvannie.springbootbookecommerce.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class UpdateUserDTO {


    @JsonProperty("username")
    private String username;

    @JsonProperty("account")
    private String account;

    @JsonProperty("email")
    private String email;

    @JsonProperty("password")
    private String password;

    @JsonProperty("facebook_account_id")
    private int facebookAccountId;

    @JsonProperty("google_account_id")
    private int googleAccountId;

    @JsonProperty("retype_password")
    private String retypePassword;

    @JsonProperty("phone_number")
    private String phoneNumber;
}
